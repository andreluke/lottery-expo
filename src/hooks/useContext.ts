import { useContext } from 'react';
import { LotteryContext } from '../contexts/LotteryContexts';

export const useLottery = () => {
  return useContext(LotteryContext);
};
