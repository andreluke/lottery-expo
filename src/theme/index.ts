// src/themes/theme.ts
import { StyleSheet } from "react-native";

export const theme = {
  colors: {
    background: "#ffffff",
    menuBackground: '#f4fcff',
    text: "#333333",
    megaSena: {
      background: "#209869",
      text: "#ffffff",
    },
    quina: {
      background: "#260085",
      text: "#ffffff",
    },
    timemania: {
      background: "#FFF600",
      text: "#049645",
    },
    loading: "#209869",
  },
  fontSizes: {
    title: 24,
    subtitle: 14,
    text: 16,
  },
  spacing: {
    small: 4,
    medium: 8,
    large: 16,
  },
  button: '#2d74da'
};

export const styles = StyleSheet.create({
  ballContainer: {
    flexDirection: "row",
  },
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  ball: {
    borderRadius: 25,
    width: 50,
    height: 50,
    justifyContent: "center",
    alignItems: "center",
    margin: theme.spacing.small,
  },
  number: {
    fontSize: theme.fontSizes.text,
    fontWeight: "bold",
  },
  subText: {
    fontSize: theme.fontSizes.subtitle,
    color: "#5d5d5d",
    marginTop: theme.spacing.small,
  },
  title: {
    fontSize: theme.fontSizes.title,
    alignSelf: "center",
    marginBottom: theme.spacing.large,
    fontWeight: "bold",
  },
});

export const style = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.colors.menuBackground,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 20,
    color: theme.colors.text,
  },
  buttonText: {
    color: theme.colors.background,
    fontSize: 16,
},
button: {
  backgroundColor: theme.button,
  paddingVertical: 10,
  borderRadius: 35,
  alignItems: 'center',
  width: '40%',
},
});