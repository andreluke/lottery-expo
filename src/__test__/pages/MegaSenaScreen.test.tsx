import React from 'react';
import renderer from 'react-test-renderer';
import MegaSenaScreen from '../../pages/MesaSenaScreen';
import { useLottery } from '../../hooks/useContext';
import { styles } from '../../theme';
import { Text } from 'react-native';

// Mock do hook useLottery
jest.mock('../../hooks/useContext', () => ({
  useLottery: jest.fn(),
}));

describe('MegaSenaScreen', () => {
  it('deve exibir o estado de carregamento corretamente', () => {
    // Mock para simular o estado de carregamento
    (useLottery as jest.Mock).mockReturnValue({
      results: null,
      loading: true,
    });

    const tree = renderer.create(<MegaSenaScreen />).toJSON();
    
    // Verifica se o snapshot do estado de carregamento está correto
    expect(tree).toMatchSnapshot();
  });

  it('deve exibir o título da Mega-Sena', () => {
    // Mock para simular resultados de Mega-Sena com o estado de carregamento falso
    (useLottery as jest.Mock).mockReturnValue({
      results: {
        megasena: {
          dezenas: [7, 12, 23, 34, 45, 56],
          dataPorExtenso: '29 de Outubro de 2024',
        },
      },
      loading: false,
    });

    const tree = renderer.create(<MegaSenaScreen />).toJSON();
    
    // Verifica se o snapshot do componente com resultados da Mega-Sena está correto
    expect(tree).toMatchSnapshot();
  });

  it('deve exibir os números corretos da loteria', () => {
    // Mock para simular resultados da Mega-Sena
    (useLottery as jest.Mock).mockReturnValue({
      results: {
        megasena: {
          dezenas: [7, 12, 23, 34, 45, 56],
          dataPorExtenso: '29 de Outubro de 2024',
        },
      },
      loading: false,
    });

    const component = renderer.create(<MegaSenaScreen />);
    const instance = component.root;

    // Verifica se os números corretos são exibidos no componente
    const ballNumbers = instance.findAllByType(Text).map((text) => text.props.children);
    
    expect(ballNumbers).toContain(7);
    expect(ballNumbers).toContain(12);
    expect(ballNumbers).toContain(23);
    expect(ballNumbers).toContain(34);
    expect(ballNumbers).toContain(45);
    expect(ballNumbers).toContain(56);
  });

  it('deve exibir a data correta do sorteio', () => {
    // Mock para simular o retorno de resultados da Mega-Sena com data
    (useLottery as jest.Mock).mockReturnValue({
      results: {
        megasena: {
          dezenas: [7, 12, 23, 34, 45, 56],
          dataPorExtenso: '29 de Outubro de 2024',
        },
      },
      loading: false,
    });

    const component = renderer.create(<MegaSenaScreen />);
    const instance = component.root;

    const dateTextArray = instance.findByProps({ style: styles.subText }).props.children;
    
    // Corrige o teste unindo os elementos do array em uma string e removendo espaços adicionais
    const dateText = dateTextArray.join('').trim();

    expect(dateText).toBe('29 de Outubro de 2024');
  });
});
