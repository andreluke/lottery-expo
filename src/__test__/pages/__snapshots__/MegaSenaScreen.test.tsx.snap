// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MegaSenaScreen deve exibir o estado de carregamento corretamente 1`] = `
<Text
  style={
    {
      "fontSize": 16,
    }
  }
>
  Carregando...
</Text>
`;

exports[`MegaSenaScreen deve exibir o t√≠tulo da Mega-Sena 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "center",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 16,
      }
    }
  >
    MEGA-SENA
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "margin": 4,
            "width": 50,
          },
          {
            "backgroundColor": "#209869",
            "margin": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#ffffff",
            },
          ]
        }
      >
        7
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "margin": 4,
            "width": 50,
          },
          {
            "backgroundColor": "#209869",
            "margin": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#ffffff",
            },
          ]
        }
      >
        12
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "margin": 4,
            "width": 50,
          },
          {
            "backgroundColor": "#209869",
            "margin": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#ffffff",
            },
          ]
        }
      >
        23
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "margin": 4,
            "width": 50,
          },
          {
            "backgroundColor": "#209869",
            "margin": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#ffffff",
            },
          ]
        }
      >
        34
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "margin": 4,
            "width": 50,
          },
          {
            "backgroundColor": "#209869",
            "margin": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#ffffff",
            },
          ]
        }
      >
        45
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "margin": 4,
            "width": 50,
          },
          {
            "backgroundColor": "#209869",
            "margin": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#ffffff",
            },
          ]
        }
      >
        56
      </Text>
    </View>
  </View>
  <Text
    style={
      {
        "color": "#5d5d5d",
        "fontSize": 14,
        "marginTop": 4,
      }
    }
  >
     
    29 de Outubro de 2024
  </Text>
</View>
`;
